src/compiler/parser/index.js

```diff
@@ -483,7 +479,13 @@ function processAttrs (el) {
       } else if (onRE.test(name)) { // v-on
         name = name.replace(onRE, '')
         addHandler(el, name, value, modifiers, false, warn)
-      } else { // normal directives
+      } else {
+        // bind update event for sync modifier and bind object
+        if (modifiers && modifiers.sync && name === 'v-bind') {
+          addSyncUpdateHandler(el, value, value)
```

Comment:
I think it's a bit more complicated than that as we need to add a handler for every property in the `v-bind` object but I haven't researched much.

Also, imagine the case where `value` is directly an object: `{ foo, bar }`, we really want to add `update:foo` and `update:bar`